src/Containers/Timeline.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useCallback } from 'react';
+import styled from 'styled-components';
+import { StyledIcon } from 'styled-icons/types';
+import { flex } from '../Utils/Mixins';
+import {
+    Responsive,
+    MainInterface,
+    ResponsiveInterface,
+} from '../Utils/BaseStyles';
+import { SmallText } from '../Text';
+import { MainTheme } from '../Themes';
+
+interface DataItem {
+    color: string;
+    Icon: StyledIcon;
+    text: string;
+}
+
+export interface TimelineProps
+    extends MainInterface,
+        ResponsiveInterface,
+        React.HTMLAttributes<HTMLDivElement> {
+    title: string;
+    titleColor: string;
+    subContentLeftTitle: string;
+    subContentLeftAmount: string | number;
+    subContentRightTitle: string;
+    subContentRightAmount: string | number;
+    figuresColor: string;
+    textColor: string;
+    separatorColor: string;
+    separatorLength: string | number;
+    data: Array<DataItem>;
+}
+
+export const Timeline: React.FC<TimelineProps> = ({
+    title,
+    titleColor,
+    subContentLeftTitle,
+    subContentLeftAmount,
+    subContentRightTitle,
+    subContentRightAmount,
+    figuresColor,
+    textColor,
+    separatorColor,
+    separatorLength,
+    data,
+    ...props
+}): React.ReactElement => {
+    const subContentLeftAmountLocale = `$${subContentLeftAmount.toLocaleString()}`;
+    const subContentRightAmountLocale = `$${subContentRightAmount.toLocaleString()}`;
+
+    const getTimelineItem = useCallback(
+        (Item: DataItem, index: number) => (
+            <TimelineItem key={`${index}_timeline`}>
+                <TimelineItemLeft color={separatorColor} isFirst={index === 0}>
+                    <Item.Icon style={{ width: '15px', color: Item.color }} />
+                </TimelineItemLeft>
+                <TimelineItemRight color={textColor}>
+                    {Item.text}
+                </TimelineItemRight>
+            </TimelineItem>
+        ),
+        [separatorColor, textColor],
```

Comment:
These dependancies look off to me. Don't you want this to change when new data is recieved?

Reply:
data is supplied to this function through an argument, being the argument each item of the data array.